From 8629fb59f9ce5a5a567eed094d88d63be398e2d0 Mon Sep 17 00:00:00 2001
From: Alexander Richards <electrodeyt@gmail.com>
Date: Sat, 13 Jun 2026 23:17:14 +0200
Subject: [PATCH] [PATCH] SVR4 mlibc stuff

---
 os/connection.c | 2 +-
 os/io.c         | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/os/connection.c b/os/connection.c
index 35e2a4d45..f441e142f 100644
--- a/os/connection.c
+++ b/os/connection.c
@@ -632,7 +632,7 @@ AllocNewConnection(XtransConnInfo trans_conn, int fd, CARD32 conn_time)
     }
     client->local = ComputeLocalClient(client);
     ospoll_add(server_poll, fd,
-               ospoll_trigger_edge,
+               ospoll_trigger_level,
                ClientReady,
                client);
     set_poll_client(client);
diff --git a/os/io.c b/os/io.c
index 83986af92..9d1d28598 100644
--- a/os/io.c
+++ b/os/io.c
@@ -358,7 +358,7 @@ ReadRequestFromClient(ClientPtr client)
         if (result <= 0) {
             if ((result < 0) && ETEST(errno)) {
                 mark_client_not_ready(client);
-#if defined(SVR4) && defined(__i386__) && !defined(__sun)
+#if defined(SVR4) && defined(__i386__) && !defined(__sun) && !defined(__mlibc__)
                 if (0)
 #endif
                 {
-- 
2.54.0

